src/tariff_region.h

```diff
@@ -0,0 +1,126 @@
+ private:
+  // Compute aggregated tariff from all importer regions along the hierarchy
+  double ComputeAggregatedTariff(cyclus::Facility* supplier, 
+                                  cyclus::Facility* requester,
+                                  const std::string& commodity);
+  
+  // Find the Lowest Common Ancestor of two regions
+  cyclus::Region* FindLowestCommonAncestor(cyclus::Region* r1, cyclus::Region* r2);
+  
+  // Get the chain of ancestors from a region up to (and including) root
+  std::vector<cyclus::Region*> GetAncestorChain(cyclus::Region* region);
+  
+  // Find the most specific tariff rule for a supplier's hierarchy from an importer's perspective
+  // Returns the tariff value, checking supplier region hierarchy from most to least specific
+  double FindMostSpecificTariff(cyclus::Region* importer_region,
+                                const std::vector<cyclus::Region*>& supplier_hierarchy,
+                                const std::string& commodity);
+  
+  // Find the appropriate tariff for a given region and commodity
+  // Uses tiered override system: region-specific commodity > region blanket > 
+  // global commodity > global blanket
+  double FindTariffForCommodity(cyclus::Region* region, const std::string& commodity);
+  
+  // Helper: Check if any tariff configuration exists
+  bool HasTariffConfiguration() const;
+  
+  // Helper: Check if configuration has been recorded
+  bool ConfigurationRecorded() const;
+  
+  // Helper: Get region prototype name (reduces repeated code)
+  std::string GetRegionName(cyclus::Region* region) const;
+  
+  // Validate the tariff configuration
+  void ValidateConfiguration();
+  
+  // Record tariff configuration to database
+  void RecordTariffConfiguration();
+
+  #pragma cyclus
+
+  // Template function to reduce code duplication between AdjustMatlPrefs and 
+  // AdjustProductPrefs
+  template<typename T>
+  void AdjustPrefsImpl(typename cyclus::PrefMap<T>::type& prefs);
+  
+  // clang-format off
+  #pragma cyclus var { \
+    "default": {}, \
+    "alias": ["adjustment_regions", "region", ["AdjustmentConfig", "blanket_adjustment", ["commodity_adjustments", "commodity", "adjustment"]]], \
+    "doc": "Tariff configuration: map from region name to (blanket_adjustment, commodity_adjustments_map). Each region can have a blanket adjustment for all commodities and specific adjustments per commodity." \
+  }
+  std::map<std::string, std::pair<double, std::map<std::string, double>>> adjustment_regions_;
+
+  #pragma cyclus var { \
+    "default": 0.0, \
+    "doc": "Optional global blanket adjustment applied to all regions and commodities." \
+  }
+  double global_blanket_adjustment;
+
+  #pragma cyclus var { \
+    "default": {}, \
+    "alias": ["global_commodity_adjustments", "commodity", "adjustment"], \
+    "doc": "Optional global commodity adjustments applied to all regions for specific commodities." \
+  }
+  std::map<std::string, double> global_commodity_adjustments_;
```

Comment:
What if these just lived in your normal `adjustment_regions_` list, but we used `"*"` for global region and `"*"` for global commodities.  Then you wouldn't need to worry about empty lists.  You could still extract and store that data in useful data structures after reading the XML (maybe in `EnterNotify()`)?